src/Map.php

```diff
@@ -3727,4 +3727,32 @@ protected function visit( iterable $entries, array &$result, int $level, ?\Closu
 			}
 		}
 	}
+
+
+	/**
+	 * Executes $callback if the Map is empty
+	 *
+	 * Examples: (Returns match or dies if doesn't exist)
+	 *  $page = $this->pages
+	 *          ->filter(fn(Page $page, $_) => $page->Id == $pageId)
+	 *          ->if(
+	 *              fn(Map $Map) => $Map->isEmpty(),
+	 *              fn(Map $_) => die("unknown page: " . $pageId))
+	 *          ->first();
+	 *
+	 * @param callable|bool $condition Bool or function with (Map) parameter which evaluates to bool
+	 * @param callable $then Function with (Map) parameter which returns void
+	 * @param callable|null $else Optional function with (Map) parameter which returns void
+	 * @return self Same map for fluid interface
+	 */
+	public function if($condition, callable $then, callable $else = null): self
+	{
+		if($condition instanceof \Closure ? $condition($this) : $condition){
+			$then($this);
+		}elseif($else != null) {
+			$else($this);
+		}
```

Comment:
Please add additional white spaces here too

Reply:
Resolved in 4e5cf55e0d52299bd0c86f499090473f0c5ab578